ARMmbed/meta-mbl
# Add default path to user supplied FIP signing key

For IOTMBL-2034: Make it easier to manually specify the key used to sign
FIP images

It wasn't obvious to me how to set the key used for signing the FIP
images. We should address that in the docs, but make it a bit more
obvious in the code by documenting the variable containing the path to
the key and giving it a default value.

**Related PRs:**
* https://github.com/ARMmbed/meta-mbl/pull/545 (forward port to warrior-dev branch).

**Testing done:**
* Full build of pelion-edge with MBL BSP.
* Test that a key at build/rot_key.pem is used by the ATF recipe:
   * `bitbake virtual/atf`
   * Get md5sum of rot_key.pem in deploy dir (e1bb5479134c24b32df5cfc98163d6f4).
   * Copy rot_key.pem to ~/tmp/rot_key.pem for safe keeping.    * `bitbake -c cleansstate virtual/atf`
   * Check that rot_key.pem doesn't exist in deploy directory
   * `bitbake virtual/atf`
   * Get md5sum of rot_key.pem in deploy dir (d35553c8032b7008d42888bbd63ff40d).
   * Copy ~/tmp/rot_key.pem to the build directory (default value of `MBL_ROT_KEY`).
   * `bitbake -c cleansstate virtual/atf`
   * `bitbake virtual/atf`
   * Get md5sum of rot_key.pem in deploy dir (e1bb5479134c24b32df5cfc98163d6f4 as expected).
* Test that overriding `MBL_ROT_KEY` in local.conf works (continued build session from above):
   * `bitbake -c cleansstate virtual/atf`
   * Delete rot_key.pem in build dir.
   * `bitbake virtual/atf`
   * Get md5sum of rot_key.pem in deploy_dir (4d2a89963aaca2224704b51cde0eef6e).
   * Override `MBL_ROT_KEY` value in local.conf:
     ```
     cat >> conf/local.conf <<EOF
     MBL_ROT_KEY = "\${TOPDIR}/alternative_rot_key_location.pem"
     EOF
     ```
   * Copy ~/tmp/rot_key.pem to alternative_rot_key_location.pem in build dir.
   * `bitbake -c cleansstate virtual/atf`
   * `bitbake virtual/atf`
   * Get md5sum of rot_key.pem in deploy dir (e1bb5479134c24b32df5cfc98163d6f4 as expected).
